src/OrchardCore.Modules/OrchardCore.Search.Elastic/Services/ElasticIndexManager.cs

```diff
@@ -0,0 +1,228 @@
+        {
+            //Convert Document to a structure suitable for Elastic
+            List<ElasticDocument> documents = new List<ElasticDocument>();
+            foreach (var indexDocument in indexDocuments)
+            {
+                documents.Add(CreateElasticDocument(indexDocument));
+            }
+            var result = await _elasticClient.IndexManyAsync(documents, indexName);
+            if (result.Errors)
+            {
+                _logger.LogWarning($"There were issues reported indexing the documents. {result.ServerError}");
+            }
+        }
+
+        public async Task<ElasticTopDocs> SearchAsync(string indexName, string query)
+        {
+            ElasticTopDocs elasticTopDocs = new ElasticTopDocs();
+            if (await Exists(indexName))
+            {
+                var searchResponse = await _elasticClient.SearchAsync<ElasticDocument>(s => s
+                    .Index(indexName)
+                    .Query(q => new RawQuery(query))
+                    );
+
+                if (searchResponse.IsValid)
+                {
+                    elasticTopDocs.Count = searchResponse.Documents.Count;
+                    elasticTopDocs.TopDocs = searchResponse.Documents.ToList();
+                }
+
+                _timestamps[indexName] = _clock.UtcNow;
+            }
+            return elasticTopDocs;
+        }
+
+        private ElasticDocument CreateElasticDocument(DocumentIndex documentIndex)
+        {
+            ElasticDocument elasticDocument = new ElasticDocument(documentIndex.ContentItemId.ToLowerInvariant());
+            foreach (var entry in documentIndex.Entries)
+            {
+                if (entry.Name.Contains("Analyzed") || entry.Name.Contains("Sanitize") || entry.Name.Contains("Normalized"))
+                {
+                    continue;
+                }
+
+                switch (entry.Type)
+                {
+                    case DocumentIndex.Types.Boolean:
+                        // store "true"/"false" for booleans
+                        elasticDocument.Set(entry.Name, (bool)(entry.Value));
+                        break;
+
+                    case DocumentIndex.Types.DateTime:
+                        if (entry.Value != null)
+                        {
+                            if (entry.Value is DateTimeOffset)
+                            {
+                                elasticDocument.Set(entry.Name, ((DateTimeOffset)(entry.Value)).UtcDateTime);
+                            }
+                            else
+                            {
+                                elasticDocument.Set(entry.Name, ((DateTime)(entry.Value)).ToUniversalTime());
+                            }
+                        }
+                        //else
+                        //{
+                        //    elasticDocument.Set(entry.Name, null);
+                        //}
```

Comment:
Can you confirm that we can use an Exists query to reach the same goal if we remove these?

Reply:
Let me first do a RCA and figure out the best solution!

Reply:
> @Sen-Gupta
> 
> Thanks for this work, it may fix one of the remaining concerns when running in a distributed environment

You guys have done some amazing work. I'm  just helping myself!

Reply:
The issue is that if we don't index a value for a Field or Part then it won't find any document when we do a wildcard query.

```json
{
  "query": {
    "wildcard": {
      "Article.TitlePart": {
        "value": "*"
      }
    }
  }
}
```

So ElasticSearch documentation recommends using a default null value for these.

https://www.elastic.co/guide/en/elasticsearch/reference/current/null-value.html